### ISSUE TITLE
Support multiple tabs per site

### ISSUE THREAD

**Issue by dahjelle** [kofigumbs/multi#57]
I've been wanting to set up a site-specific browser for several web apps I use regularly: Google Docs, GitLab, etc. However, I often end up opening multiple pages on that site in multiple tabs, such as multiple docs I'm working on or an issue and its related pull request. At the moment, it appears that Multi's tabs are only for _separate_ sites, not different instances of the _same_ site. Is that true, or is that a configuration option I missed?

Otherwise, I'd like to request that as a feature!

**Comment by kofigumbs** [comment]
You can use multiple Multi tabs for the same site using the JSON config. For example:

```json
{
  "tabs": [
    { "title": "Code", "url": "https://gitlab.com/org/repo" },
    { "title": "Issues", "url": "https://gitlab.com/org/repo/issues" }
  ]
}
```

I suspect the number of sites may not be the root issue though—it sounds like you're interested in creating tabs as you work. Multi doesn't support any dynamic tab creation right now. The only tabs you get are the ones you list in the JSON. Happy to keep this issue open with the enhancement tag, but I don't imagine prioritizing it soon. At the point where you need dynamic window management, I don't see why Multi would be better than using a web browser.

Lemme know if neither of those interpretations of your request are right 😅

**Comment by dahjelle** [comment]
> I suspect the number of sites may not be the root issue though—it sounds like you're interested in creating tabs as you work.

Yes, that's exactly it. Perhaps I'm digging through GitHub issues, and have 3 issues that might be related to my search. I'd (personally) open them in new tabs, and then close the tabs after I've processed them, rather than clicking on each individual issue, and then going back. (If that made any sense…)

> At the point where you need dynamic window management, I don't see why Multi would be better than using a web browser.

In my case, it helps a lot to keep browser tabs under control. Once-upon-a-time, I had a workflow (using [Fluid.app](https://fluidapp.com), which doesn't appear to be actively developed anymore…) where I would have separate apps for GitLab, Google Docs, and an internal company wiki — and I really liked reserving the browser for browsing web **pages** and using site-specific browsers for **apps**. (Admittedly, it can be hard to draw that line sometimes these days…)

Anyhow, I definitely get that I might be very unusual in my use case — but I would love to have that workflow available again!

Thanks for your time & have a great day!

**Comment by toomim** [comment]
Yeah, one way of seeing my use-case for multi is to "have a separate icon in my dock for a website."

In a sense, this is what an electron app does for you, too, with the addition that electron also lets you run node v8 code with elevated permissions.

Multi seems to be striking a sweet spot in between an electron app and a web browser.

**Comment by toomim** [comment]
I just did a little searching, and found this similar project for the paradigm of "creating a new app icon for a browser" rather than multi's "creating an app that supports multiple browser webviews": https://github.com/dmarmor/epichrome

It seems that these two paradigms are not so far apart in practice!

**Comment by voltechs** [comment]
Plus One for this feature, please!

So close to this (Multi) being exactly what I've been looking for for ages! Was using "Nativefier" to attempt to solve this for some time, but it created such heavy applications because the Electron library was not shared—it was bundled with each app.

Would love to donate/support the project. I'm a software engineer but I don't have a ton of (let alone any recent) experience developing for macOS. Would be willing to help where I can. I went on a bit of a journey before finding this issue (I promise, I tried to search first, before creating my own), so mine (https://github.com/kofigumbs/multi/issues/122) is now a duplicate. Apologies for the clutter!
